cli/lsp/diagnostics.rs

```diff
@@ -313,9 +374,14 @@ impl DiagnosticsServer {
       performance,
       ts_server,
       batch_counter: Default::default(),
+      state: Default::default(),
     }
   }
 
+  pub fn state(&self) -> Arc<RwLock<DiagnosticsState>> {
```

Comment:
Instead of having methods where a struct is received, this could instead be passed as a constructor dependency to `DiagnosticsServer`.